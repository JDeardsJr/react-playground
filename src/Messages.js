import React from 'react';
import './Messages.css';

export default function Messages(props) {
    const badge = props.unread
        ? <div className="unread_count badege">{props.unread}</div>
        : null;
    return (
        <div className="messages">
            {props.name}
            {badge}
        </div>
    );
}
