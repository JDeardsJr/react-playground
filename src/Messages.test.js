import React from 'react';
import ReactDOM from 'react-dom';
import Messages from './Messages';
import renderer from 'react-test-renderer';

describe('Messages component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Messages />, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    
    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<Messages name="Messages" unread={4} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    
    it('renders without crashing', () => {
        const tree = renderer
            .create(<Messages name="messages" unread={0} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});