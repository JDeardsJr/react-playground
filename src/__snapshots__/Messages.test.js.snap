// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messages component renders the UI as expected 1`] = `
<div
  className="messages"
>
  Messages
  <div
    className="unread_count badege"
  >
    4
  </div>
</div>
`;

exports[`Messages component renders without crashing 1`] = `
<div
  className="messages"
>
  messages
</div>
`;

exports[`renders the UI as expected 1`] = `
<div
  className="messages"
>
  Messages
  <div
    className="unread_count badege"
  >
    4
  </div>
</div>
`;

exports[`renders without crashing 1`] = `
<div
  className="messages"
>
  messages
</div>
`;
